### DISCUSSION TITLE
Conditional CJS

### DISCUSSION THREAD

**Discussion by webistomin** [frehner/modern-guide-to-packaging-js-library#30] [Q&A] [answered]
Hi! 👋

Can someone explain why we need to split the `CJS` module into production and development?

And why don't we do the same with `ESM`?

**Comment by frehner** [comment, marked as answer]
>Can someone explain why we need to split the CJS module into production and development?

You don't necessarily need to; the document mentions that it's useful if you have additional logic and/or warnings for development mode, then it's useful. But if you don't, then there's no need.

>And why don't we do the same with ESM?

I believe the document doesn't make the distinction between CJS and ESM for this; you should be doing the same thing for ESM. Is there some wording in there that lead you to not think this?

**Reply by webistomin** [reply]
Thanks for answering. Perhaps I misunderstood you.

I remember that once I bundled a library via [tsdx](https://github.com/jaredpalmer/tsdx#development-only-expressions--treeshaking), there was a `CJS`-only dev/prod split. I thought maybe there was some hidden meaning in not splitting the ESM file.

```
.
|____index.d.ts
|____index.js
|____mylib.cjs.development.js
|____mylib.cjs.development.js.map
|____mylib.cjs.production.min.js
|____mylib.cjs.production.min.js.map
|____mylib.esm.js
|____mylib.esm.js.map
```

**Reply by frehner** [reply]
Seems like something you could talk to that bundler about. 👍
